### PR TITLE
feat(gatsby): Track persistence time of save-state

### PR DESCRIPTION
**Pull request by abhiaiyer91**
Tracking duration of save-state.

[ch32214]

### CODE REVIEW

File: packages/gatsby/src/redux/save-state.js [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 async function saveState() {
   if (saveInProgress) return
   saveInProgress = true
 
+  const startTime = new Date()
+
   try {
     await redux.saveState()
+
+    const duration = (new Date().getTime() - startTime.getTime()) / 1000
```
Comment (jamo) [inline packages/gatsby/src/redux/save-state.js:16]: This needs to be an Integer

Comment (abhiaiyer91) [reply packages/gatsby/src/redux/save-state.js:16]: How do we track duration in other places? Ill follow

File: packages/gatsby/src/redux/save-state.js [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
 const report = require(`gatsby-cli/lib/reporter`)
+const { captureEvent } = require(`gatsby-telemetry`)
 const redux = require(`./`)
 
 let saveInProgress = false
 async function saveState() {
   if (saveInProgress) return
   saveInProgress = true
 
+  const startTime = new Date()
```
Comment (KyleAMathews) [inline packages/gatsby/src/redux/save-state.js:11]:
```suggestion
  const startTime = Date.now()
```

File: packages/gatsby/src/redux/save-state.js (hunk 1 above)
Comment (KyleAMathews) [reply packages/gatsby/src/redux/save-state.js:20]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime returns the same thing as `Date.now()`

Comment (abhiaiyer91) [reply packages/gatsby/src/redux/save-state.js]: used Date.now

**Comment by abhiaiyer91** [comment]
Are there any risks with using `captureEvent` which was introduced in a later version of telemetry @jamo @KyleAMathews

**Review by jamo — APPROVED** [review]

**Comment by KyleAMathews** [comment]
Just check that the version of `gatsby-telemetry` that `gatsby` depends on includes `captureEvent`. If it doesn't, bump it to one that does (so just the latest).

**Comment by abhiaiyer91** [comment]
> Just check that the version of `gatsby-telemetry` that `gatsby` depends on includes `captureEvent`. If it doesn't, bump it to one that does (so just the latest).

Nice, looks like its on latest

**Comment by LekoArts** [comment]
As mentioned in the ticket by Vlad:

> But one thing to keep in mind is that we actually save the state multiple times during one build.

So we probably only want to send this telemetry event once.

**Comment by jamo** [comment]
We can always pick the final event, and maybe it's also useful to know how expensive the save is and know how long overall it takes?

**Review by jamo — APPROVED** [review]

**Pull request merged**